client/src/components/Project/Project.js

```diff
@@ -7,10 +7,10 @@ const Project = props => {
       <div>description: {props.description}</div>
       <div>username: {props.username}</div>
       <div>votes: {props.votes}</div>
-      <button onClick={event => props.handleUpVote(event, props.id)}>
+      <button onClick={event => props.hasUserVotedOnThisProject(event, props.id, "upVote")}>
         Up
       </button>
-      <button onClick={event => props.handleDownVote(event, props.id)}>
+      <button onClick={event => props.hasUserVotedOnThisProject(event, props.id, "downVote")}>
```

Comment:
Both upvote and downvote buttons will first check of the user has voted on this project. To tell the difference between upvote and downvote buttons, I have passed in a string to differentiate the two.

Reply:
So the downvote is set & the upvote needs to be implemented..

Reply:
They are both implemented. Instead of executing the queries directly, both buttons will call the hasUserVotedOnThisProject function to check if they voted on the project before. This is where the code decides what to do next.